Examples/test-suite/fortran_naming.i

```diff
@@ -0,0 +1,79 @@
+%module fortran_naming
+
+%warnfilter(SWIGWARN_LANG_IDENTIFIER);
+
+%fortran_bindc_struct(MyStruct);
+
+// Without this line, SWIG should raise an error rather than propagating the
+// invalid name into the Fortran code.
+%rename(FooClass) _Foo;
```

Comment:
Making SWIG error out because of a naming problem is unprecedented. This should be ignored with a warning or automatically renamed (like Ruby).

Reply:
@wsfulton 
Ruby has a very limited renaming capability: it only renames constant identifiers, which (I think?) aren't ever referenced by wrapped code after that point. This differs from symbols like classes, which can be referenced (as function parameters) even before they're declared to SWIG.

The options for dealing with "_"-prefixed identifiers that we have (at least so far as I've tried) are:
1. Don't do any renaming and suffer compile-time errors in the fortran code.
2. Require PCRE support so we can use a regex to match names that start with `_`.
3. Add a custom match keyword like `prefixtarget`.
4. Rename identifiers when they're being wrapped (as Ruby does, though just for constants), but risk compile-time errors if those identifiers are used anywhere else.
5. Rename locally-scoped identifiers when possible, but raise errors when identifiers that can be referenced elsewhere by Fortran-wrapped code (modules, classes).

I'm not sure what else I can do.

Reply:
It took a lot of effort, but I've implemented an automatic renaming scheme.